File: training task/3_multilevel_inheritance.py
class Person:
    def __init__(self, name, age):
        self.name = name
        self.age = age

    def get_person_details(self):
        return f"Name: {self.name}, Age: {self.age}"

class Student(Person):
    def __init__(self, name, age, student_id):
        super().__init__(name, age)
        self.student_id = student_id

    def get_student_details(self):
        person_details = self.get_person_details()
        return f"{person_details}, Student ID: {self.student_id}"

class GraduateStudent(Student):
    def __init__(self, name, age, student_id, thesis_title):
        super().__init__(name, age, student_id)
        self.thesis_title = thesis_title

    def get_graduate_student_details(self):
        student_details = self.get_student_details()
        return f"{student_details}, Thesis Title: {self.thesis_title}"

grad_student = GraduateStudent("Mou", 23, "G43", "Machine Learning in Healthcare")
print(grad_student.get_graduate_student_details())

File: training task/3_multilevel_inheritance.cpp
#include <iostream>
#include <string>

using namespace std;
class Person {
protected:
    string name;
    int age;

public:
    Person(string name, int age) {
        this->name = name;
        this->age = age;
    }

    string getPersonDetails() const {
        return "Name: " + name + ", Age: " + to_string(age);
    }
};

class Student : public Person {
private:
    string student_id;

public:
    Student(string name, int age, string student_id) : Person(name, age) {
        this->student_id = student_id;
    }
    string getStudentDetails() const {
        return getPersonDetails() + ", Student ID: " + student_id;
    }
};

class GraduateStudent : public Student {
private:
    string thesis_title;

public:
    GraduateStudent(string name, int age, string student_id, string thesis_title)
        : Student(name, age, student_id) {
        this->thesis_title = thesis_title;
    }
    string getGraduateStudentDetails() const {
        return getStudentDetails() + ", Thesis Title: " + thesis_title;
    }
};

int main() {
    GraduateStudent grad_student("Mou", 25, "G12345", "Machine Learning in Healthcare");
    cout << grad_student.getGraduateStudentDetails() << endl;

    return 0;
}
